crates/wasi-common/src/sys/unix/path.rs

```diff
@@ -204,3 +205,54 @@ pub(crate) fn remove_directory(dirfd: &OsDir, path: &str) -> Result<()> {
     unsafe { unlinkat(dirfd.as_raw_fd(), path, AtFlag::REMOVEDIR)? };
     Ok(())
 }
+
+pub(crate) fn filestat_get_at(dirfd: &OsDir, path: &str) -> Result<types::Filestat> {
+    use yanix::file::{fstatat, AtFlag};
+    // We always set `lookupflags` to SYMLINK_NOFOLLOW (0x0) since symlink expansion
+    // is taken care for us in `path::get` a level above.
+    let stat = unsafe { fstatat(dirfd.as_raw_fd(), path, AtFlag::empty())? };
+    let stat = stat.try_into()?;
```

Comment:
It's confusing, because there's both `AT_SYMLINK_FOLLOW` and `AT_SYMLINK_NOFOLLOW`, and they're both non-zero flags, because some system calls default to following by default, while others don't. It looks like `fstatat` defaults to following symbolic links, so it looks like you should need to pass `AtFlag::SYMLINK_NOFOLLOW` here.

Reply:
Oh, sorry, I meant `AT_SYMLINK_NOFOLLOW` in the comment above. I don't think we should specify any flags in the call to `fstatat` here as otherwise we'll end up with `EINVAL` since `path` will always be resolved to an actual path by now. Symlink dereferencing happens in `path::get` call level above. Does that make sense?

Reply:
Hmm, on second thought, we should pass flags here indeed. I think there some other subtlety going on here that dereferences the symlink too far, or something like that. I'll need to look closer and see what's happening here under the hood.

Reply:
OK, you're absolutely right we should pass appropriate flags to `fstatat`. I misread what you wrote (I guess tiredness is winning today), and missed the fact that `fstatat` *always* dereferences unless we specify `AT_SYMLINK_NOFOLLOW`.